src/Raven.Server/Documents/Indexes/Persistence/Corax/JintCoraxDocumentConverter.cs

```diff
@@ -359,37 +367,49 @@ private static JsValue TryDetectDynamicFieldCreation(string property, ObjectInst
                 var propertyNameAsString = propertyName.AsString();
                 if (string.Equals(propertyNameAsString, nameof(CreateFieldOptions.Indexing), StringComparison.OrdinalIgnoreCase))
                 {
-                    field.Indexing = GetEnum<FieldIndexing>(optionValue, propertyNameAsString);
+                    fieldIndexing = GetEnum<FieldIndexing>(optionValue, propertyNameAsString);
 
                     continue;
                 }
 
                 if (string.Equals(propertyNameAsString, nameof(CreateFieldOptions.Storage), StringComparison.OrdinalIgnoreCase))
                 {
                     if (optionValue.IsBoolean())
-                        field.Storage = optionValue.AsBoolean()
+                        fieldStorage = optionValue.AsBoolean()
                             ? FieldStorage.Yes
                             : FieldStorage.No;
                     else
-                        field.Storage = GetEnum<FieldStorage>(optionValue, propertyNameAsString);
+                        fieldStorage = GetEnum<FieldStorage>(optionValue, propertyNameAsString);
 
                     continue;
                 }
 
                 if (string.Equals(propertyNameAsString, nameof(CreateFieldOptions.TermVector), StringComparison.OrdinalIgnoreCase))
                 {
-                    field.TermVector = GetEnum<FieldTermVector>(optionValue, propertyNameAsString);
+                    //ignore
+                    //field.TermVector = GetEnum<FieldTermVector>(optionValue, propertyNameAsString);
 
                     continue;
                 }
             }
         }
 
-        if (scope.DynamicFields.ContainsKey(field.Name) == false)
+        scope.DynamicFields ??= new();
+        if (scope.DynamicFields.TryGetValue(fieldName, out var persistedIndexField))
+        {
+            if (persistedIndexField.Indexing != fieldIndexing)
+                throw new InvalidDataException($"Inconsistent dynamic field creation options were detected. Field '{fieldName}' was created with '{persistedIndexField.Indexing}' analyzer but now '{field.Indexing}' analyzer was specified. This is not supported");
```

Comment:
Shouldn't we throw only if an index version is `JavaScriptProperlyHandleDynamicFieldsIndexFields` ?

Reply:
ping